pkg/sink/sink_test.go

```diff
@@ -141,25 +109,57 @@ func getSinkAssets(t *testing.T, resources test.Resources, elName string, auth A
 	return r, dynamicClient
 }
 
-// getCreatedPipelineResources returns the pipeline resources that were created from the given actions
-func getCreatedPipelineResources(t *testing.T, actions []ktesting.Action) []pipelinev1alpha1.PipelineResource {
+// setupInterceptors creates a httptest server with all coreInterceptors and any passed in webhook interceptor
+// It returns a http.Client that can be used to talk to these interceptors
+func setupInterceptors(t *testing.T, k kubernetes.Interface, l *zap.SugaredLogger, webhookInterceptor http.Handler) *http.Client {
 	t.Helper()
-	prs := []pipelinev1alpha1.PipelineResource{}
+	// Setup a handler for core interceptors using httptest
+	coreInterceptors, err := server.NewWithCoreInterceptors(k, l)
+	if err != nil {
+		t.Fatalf("failed to initialize core interceptors: %v", err)
+	}
+	rtr := mux.NewRouter()
+	// server core interceptors by matching on req host
+	rtr.MatcherFunc(func(r *http.Request, _ *mux.RouteMatch) bool {
+		return strings.Contains(r.Host, interceptors.CoreInterceptorsHost)
+	}).Handler(coreInterceptors)
+
+	if webhookInterceptor != nil {
+		rtr.Handle("/", webhookInterceptor)
+	}
+	srv := httptest.NewServer(rtr)
+	t.Cleanup(func() {
+		srv.Close()
+	})
+	httpClient := srv.Client()
+	u, err := url.Parse(srv.URL)
+	if err != nil {
+		t.Fatalf("testServer() url parse err: %v", err)
+	}
+	httpClient.Transport = &http.Transport{
+		Proxy: http.ProxyURL(u),
+	}
+	return httpClient
+}
+
+// toTaskRun returns the task run that were created from the given actions
+func toTaskRun(t *testing.T, actions []ktesting.Action) []pipelinev1.TaskRun {
+	t.Helper()
+	trs := []pipelinev1.TaskRun{}
 	for i := range actions {
 		obj := actions[i].(ktesting.CreateAction).GetObject()
 		// Since we use dynamic client, we cannot directly get the concrete type
 		uns := obj.(*unstructured.Unstructured).Object
-		pr := pipelinev1alpha1.PipelineResource{}
-		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(uns, &pr); err != nil {
+		tr := pipelinev1.TaskRun{}
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(uns, &tr); err != nil {
 			t.Errorf("failed to get created pipeline resource: %v", err)
 		}
-		prs = append(prs, pr)
+		trs = append(trs, tr)
 	}
-	return prs
+	return trs
 }
 
-// checkSinkResponse checks that the sink response status code is 201
-// and that the body returns the EventListener, namespace, and eventID.
+// checkSinkResponse checks that the sink response status code is 201 and that the body returns the EventListener, namespace, and eventID.
```

Comment:
Quick question: Why was this put into one line? Are there any coding guidelines for comments?

I usually see comments split consistently but here it was reverted.

Reply:
yeah this looks inadvertent. will revert.